autoload/committia/git.vim

```diff
@@ -42,6 +42,11 @@ function! s:extract_first_line(str) abort
 endfunction
 
 function! s:search_git_dir_and_work_tree() abort
+    " Use environment variables if set
+    if !empty($GIT_DIR) && !empty($GIT_WORK_TREE)
+        return [$GIT_DIR, $GIT_WORK_TREE]
+    endif
```

Comment:
`$GIT_WORK_TREE` is set by user and it must be existing. So `$GIT_WORK_TREE` should be verified as follows:

```vim
if !empty($GIT_DIR) && !empty($GIT_WORK_TREE) && isdirectory($GIT_WORK_TREE)
    return [$GIT_DIR, $GIT_WORK_TREE]
endif
```

Reply:
Or should we explicitly raise an error rather than falling back into automatic worktree detection silently?

```vim
if !empty($GIT_DIR) && !empty($GIT_WORK_TREE)
    if !isdirectory($GIT_WORK_TREE)
        throw 'Directory specified with $GIT_WORK_TREE is not found: ' . $GIT_WORK_TREE
    endif
    return [$GIT_DIR, $GIT_WORK_TREE]
endif
```

Either look ok for me.

Reply:
Good point, I set `GIT_WORK_TREE` manually, that's correct. I like the variant throwing an error because git cannot operate if the value of `GIT_WORK_TREE` is invalid:
```bash
✔ test master@HEAD$ GIT_WORK_TREE="/tmp/invalid" git status
fatal: This operation must be run in a work tree
```